projects/js-packages/components/components/alert/index.tsx

```diff
@@ -43,9 +43,13 @@ const getIconByLevel = level => {
  * @param {string} props.level             - The alert level: error, warning, info, success.
  * @param {boolean} props.showIcon         - Whether to show the alert icon.
  * @param {React.Component} props.children - The alert content.
- * @returns {React.Component}                The `Alert` component.
+ * @returns {React.ReactElement}             The `Alert` component.
  */
-function Alert( { level, children, showIcon } ) {
+const Alert: React.FC< React.PropsWithChildren< AlertProps > > = ( {
```

Comment:
I'm afraid children is option even after these changes :-(

<img width="352" alt="image" src="https://user-images.githubusercontent.com/77539/166737941-cff4604a-790c-4849-b95f-dbf00b1a1bbf.png">

Reply:
> I'm afraid children is option even after these changes

In JS files, you can add a comment at the top of the file

```js
// @ts-check
```

More details here
https://www.typescriptlang.org/docs/handbook/intro-to-js-ts.html#ts-check

Reply:
Another reason may be because of differing `@types/react` versions. In v18, `children` prop is no longer a part of `React.FC` while in older versions it's there in React.FC

Reply:
This should be resolved when `@types/react` is updated to v18 throughout the monorepo.

Reply:
Exactly what @manzoorwanijk said, it's being required now, but you need to opt-in for checks at `JS` files.

Reply:
> > I'm afraid children is option even after these changes

> In JS files, you can add a comment at the top of the file

The thing is, the children is optional, even removing the `PropsWithChildren`. That's why I pasted the screenshot above (it is a `.ts` file). I'd like to make it required. Adding the `children` prop to the type makes it work.

Reply:
If you go to `stories/index.jsx` and put `// @ts-check` there, you'll see that it asks for `children` and `level`:

![image](https://user-images.githubusercontent.com/1663717/166743099-af15a92a-1713-4d71-b54d-fe00f86ffc0a.png)

Reply:
it only works when ...

```ts
type AlertProps = {
	// ...
	/** Children to be rendered inside the alert. */
	children: React.ReactNode;
};
```